docs/lab/08/index.md

```diff
@@ -0,0 +1,435 @@
+---
+sidebar_position: 8
+description: Wi-Fi
+slug: /lab/08
+---
+
+# 08 - Wi-Fi
+
+This lab will teach you the basics of networking and Wi-Fi, and how to use Wi-Fi with the Raspberry Pi Pico W, in Embassy.
+
+## Resources
+
+**Andrew Tanenbaum**, *Computer networks (5th edition)*
+  - Chapter 1 - *Introduction*
+    - Subchapter 1.4.2 - *The TCP/IP Reference Model*
+
+## Networking basics
+
+A network is a set of devices that are linked together in order to share resources. Devices can talk to one another using different communication protocols, usually Ethernet (wired) or Wi-Fi (wireless).
+
+Devices that communicate in a network have an associated IP (Internet Protocol) address. There are two types of IP addresses: IPv4 (stored on 32 bits) and IPv6 (newer, stored on 64 bits). We will be discussing only the IPv4 in this lab, since it is most commonly used in local networks.
+
+### IPv4
+
+An IPv4 address looks like this:
+`192.168.0.1/24`
+Each number separated by a dot is represented on 8 bits (ranging from 0 to 255). The number after the slash is the *network mask*, which will be described shortly.
+
+An IP address is made of two parts:
+- network prefix: defines the network the device is a part of
+- host identifier: is unique to the device in the network
+
+The *network mask* defines the number of bits that are used to represent the network prefix. In other words, the smaller the mask, the more unique devices can exist in the network.
+
+For example, for the above address `192.168.0.1/24`, we know that the network prefix is 24 bits long. Therefore, the first 24 bits give us the network prefix and the last 8 bits give us the host identifier. The mask in this case would look like this: `255.255.255.0`.
+
+:::info
+In binary, `255.255.255.0` is `11111111.11111111.11111111.00000000`. 
+
+First 24 bits are 1, last 8 bits are 0. 
+:::
+
+Any device that wants to communicate in this local network (or *subnet*) must have an address ranging from `192.168.0.1` to `192.168.0.254`.
```

Comment:
Explain here that the first 24 bits are the same. You can use wavedrom to draw the address and have different colours for the bits.

```
{
    "reg": [
        {
            "bits": 5,
            "name": "Recipient",
            "type": 7,
            "attr": [
                "00000 - Device",
                "00001 - Interface",
                "00010 - Endpoint",
                "00011 - Other",
                "00100 - 11111 - Reserved"
            ]
        },
        {
            "bits": 2,
            "name": "Type",
            "type": 5,
            "attr": [
                "00 - Standard",
                "01 - Class",
                "10 - Vendor",
                "11 - Reserved"
            ]
        },
        {
            "bits": 1,
            "name": "Direction",
            "type": 3,
            "attr": [
                "0 - Host to Device",
                "1 - Device to Host"
            ]
        }
    ],
    "config": {
        "hspace": 950
    }
}

```